pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1469,9 +1478,16 @@ func TestPreFilterState(t *testing.T) {
 			p.(*PodTopologySpread).enableMatchLabelKeysInPodTopologySpread = tt.enableMatchLabelKeys
 
 			cs := framework.NewCycleState()
-			if _, s := p.(*PodTopologySpread).PreFilter(ctx, cs, tt.pod); !s.IsSuccess() {
+			_, s := p.(*PodTopologySpread).PreFilter(ctx, cs, tt.pod)
+			if tt.wantPrefilterStatus != nil && !tt.wantPrefilterStatus.IsSuccess() {
+				if diff := cmp.Diff(tt.wantPrefilterStatus, s, cmpOpts...); diff != "" {
+					t.Errorf("PodTopologySpread#PreFilter() returned diff (-want,+got):\n%s", diff)
+				}
+				return
+			} else if !s.IsSuccess() {
 				t.Fatal(s.AsError())
 			}
```

Comment:
Shouldn't only ↓ be enough to compare the status from prefilter? 

```
if diff := cmp.Diff(tt.wantPrefilterStatus, s, cmpOpts...); diff != "" {
					t.Errorf("PodTopologySpread#PreFilter() returned diff (-want,+got):\n%s", diff)
}
```

Reply:
To be honest, I decided to go this way because the number of codes to be changed in the test would be a bit large.